src/core/featuremodel.cpp

```diff
@@ -351,58 +351,147 @@ void FeatureModel::featureAdded( QgsFeatureId fid )
   setFeature( mLayer->getFeature( fid ) );
 }
 
-void FeatureModel::create()
+bool FeatureModel::create()
 {
   if ( !mLayer )
-    return;
-
-  startEditing();
+    return false;
 
-  connect( mLayer, &QgsVectorLayer::featureAdded, this, &FeatureModel::featureAdded );
-  if ( !mLayer->addFeature( mFeature ) )
+  if ( ! startEditing() )
   {
-    QgsMessageLog::logMessage( tr( "Feature could not be added" ), "QField", Qgis::Critical );
+    QgsMessageLog::logMessage( tr( "Cannot start editing" ), "QField", Qgis::Warning );
```

Comment:
Maybe you could improve these log messages by adding additional info like "Cannot start editing on layer <name> to create feature" etc. 
Just an input. Not a requested change.

Reply:
I ws thinking about it, but I was not sure how detailed I should make the log messages. Thanks for the feedback.